pkg/apis/openstackproviderconfig/v1alpha1/types.go

```diff
@@ -75,6 +81,13 @@ type NetworkParam struct {
 	Filter Filter `json:"filter,omitempty"`
 }
 
+type TagParam struct {
```

Comment:
I like this `TagParam` here and it'd be great if we could use it for the cluster-level resources too. We've recently added tags to networks created by the cluster controller in #178 it'd be great to update that code to use this TagParam and allow for custom tags to be added.

This said, I think we should have a default, tho. A tag that can't be overwritten by the custom tags set in the Cluster/Machine spec. I'm happy to keep using `cluster-api-provider-openstack=<cluster-name>` for this. Thoughts?

/cc @chrigl

Reply:
I like the idea of having a way to add custom tags to network resources, and I am with @flaper87  that we might want to have defaults that are always applied. But from the users perspective, I think we should leave `cluster-api-provider-openstack` and the `<cluster-name>` be two separate tags. So it would be more convenient to filter resources. e.g. such things would be possible

```
# all resources owned by cluster-api-provider openstack
openstack network list --tags cluster-api-provider-openstack

# all owned resources and cluster-name = default/test1
openstack network list --tags cluster-api-provider --tags default/test1

# all owned resources but *not* cluster-name = default/test1
openstack network list --tags cluster-api-provider-openstack --not-tags=default/test1
```

Reply:
Ok, so I'll keep those 2 and add the extra ones (if existing)